addon/initializers/route-alias.js

```diff
@@ -15,6 +15,16 @@ export function initialize(application) {
     }
   });
 
+  Ember.Router.reopen({
+    willTransition(a, b, intent) {
+      if (intent.targetName.indexOf('replace-with:') === 0) {
+        let terminal = intent.targetName.substr(13);
+        this.replaceWith(terminal);
+      }
+      this._super.apply(this, arguments);
```

Comment:
I think we might be able to get away with manually passing forward the new `transition` from line 22 into `_super` as long as we're 100% certain that this is the first `reopen`.

/cc @nickiaconis